kotlin-native/NEW_MM.md

```diff
@@ -122,65 +116,52 @@ kotlin.targets.withType(KotlinNativeTarget::class.java) {
-  on another, by the end the object will be frozen, but some subgraph of it might be not.
+* Compiler caches are not supported, so the compilation of debug binaries will be slower.
+* Freezing machinery is not thread-safe: if an object is being frozen on one thread, and its subgraph is being modified on another, by the end, the object will be frozen, but some subgraph of it might be not.
 * Documentation is not updated to reflect changes for the new MM.
-* There's no handling of application state on iOS: if application goes into the background, the collector will not be
-  throttled down; on the other hand the collection is not forced upon going into the background, which leaves
-  the application with a larger memory footprint than necessary, making it a more likely target to be terminated by the OS.
-* WASM (or indeed any target that does not have pthreads) is not supported with the new MM.
+* There's no application state handling on iOS: the collector will not be throttled down if the application goes into the background.
+  However, the collection is not forced upon going into the background, which leaves the application with a larger memory footprint than necessary, making it a more likely target to be terminated by the OS.
+* WASM (or any target that doesn't have threads) is not supported with the new MM.
+
 
 ## Workarounds
 
 ### Unexpected object freezing
-Some libraries might not be ready for the new memory model and freeze-transparency of `kotlinx.coroutines`, so unexpected `InvalidMutabilityException` 
-or `FreezingException` might appear.
-To workaround such cases, we added a `freezing` binary option that disables freezing fully (`disabled`) or 
-partially (`explicitOnly`).
-The former disables freezing mechanism at runtime (thus, making it no-op) while the latter disables automatic freezing of 
-[`@SharedImmutable`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native.concurrent/-shared-immutable/) globals, 
-but keeps direct calls to [`freeze`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native.concurrent/freeze.html) fully functional. 
+Some libraries might not be ready for the new memory model and freeze-transparency of `kotlinx.coroutines`, so unexpected `InvalidMutabilityException` or `FreezingException` might appear.
+
+To workaround such cases, we added a `freezing` binary option that disables full (`disabled`) or partial (`explicitOnly`) freeze.
+The former disables the freezing mechanism at runtime (thus, making it a no-op), while the latter disables automatic freezing of
+[`@SharedImmutable`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native.concurrent/-shared-immutable/) globals, but keeps direct calls to [`freeze`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native.concurrent/freeze.html) fully functional.
+
 Similar to the [new memory model](#switch-to-the-new-memory-model), there are several ways to enable this option:
-* In `gradle.properties`: 
```

Comment:
Similar to the above, maybe we should unify how compiler flags are presented here?